File: frontend/src/app/models/plan.ts
export interface Plan {
  id: number;
  name: string;
  startDate: string;
  endDate: string;
  totalBudget: number;
  totalPlanned: number;
}
export interface PlanRequest {
  name: string;
  startDate: string;
  endDate: string;
  totalBudget: number;
}

File: backend/expense_tracker_backend/src/main/kotlin/mk/sorsix/com/expense_tracker_backend/domain/Plan.kt
package mk.sorsix.com.expense_tracker_backend.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.math.BigDecimal
import java.time.LocalDate

@Entity
data class Plan(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: User = User(),
    val name: String = "",
    val startDate: LocalDate = LocalDate.now(),
    val endDate: LocalDate = LocalDate.now(),
    val totalBudget: BigDecimal = BigDecimal.ZERO,
)
